addons/sale_project/models/project.py

```diff
@@ -139,6 +139,35 @@ def _compute_display_sales_stat_buttons(self):
         for project in self:
             project.display_sales_stat_buttons = project.allow_billable and project.partner_id
 
+    def _ensure_sale_order_linked(self, sol_ids):
+        """ Orders created from project/task are supposed to be confirmed to match the typical flow from sales, but since
+        we allow SO creation from the project/task itself we want to confirm newly created SOs immediately after creation.
+        However this would leads to SOs being confirmed without a single product, so we'd rather do it on record save.
+        """
+        quotations = self.env['sale.order.line']._read_group(
+            domain=[('state', '=', 'draft'), ('id', 'in', sol_ids)],
+            aggregates=['order_id:recordset'],
+        )[0][0]
+        if quotations:
+            quotations.action_confirm()
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        projects = super().create(vals_list)
+        sol_ids = [
+            vals['sale_line_id']
+            for vals in vals_list
+            if vals.get('sale_line_id')
+        ]
+        projects._ensure_sale_order_linked(sol_ids)
```

Comment:
```suggestion
        if sol_ids:
            projects._ensure_sale_order_linked(sol_ids)
```

to only call the method once it is needed. 🙂